### PR TITLE
GAWB-480: use new submissionsCount endpoint for workspace summary page

### PR DESCRIPTION
**Pull request by davidangb**
use lightweight submissionsCount endpoint instead of heavyweight submissions endpoint.
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] **Submitter**: Tell tech lead that the PR exists if s/he wants to look at it
- [x] **Submitter**: Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [ ] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x] **Tech lead**: sign off
- [x] **LR**: sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't 
  working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev server still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/tab.cljs [hunk 1]
```diff
@@ -172,14 +172,13 @@
          (get-in ws ["workspace" "bucketName"])])
       (style/create-section-header "Analysis Submissions")
       (style/create-paragraph
-        (let [count-all (count submissions)
-              grouped-by-status (group-by #(% "status") submissions)]
+        (let [count-all (reduce + 0 (vals submissions-count))]
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/tab.cljs]: `(reduce + 0 some-list)` is effectively equivalent to `(apply + some-list)`.  It works just fine if the list is empty or nil.  `apply` should be a hair faster, but it doesn't matter.  (shrug emoji)

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/tab.cljs [hunk 2]
```diff
@@ -172,14 +172,13 @@
...
              [:div {}
               (str count-all " Submission" (when-not (= 1 count-all) "s"))
               (when (pos? count-all)
                     [:ul {:style {:marginTop "0"}}
-                     (for [[status subs] grouped-by-status]
-                        [:li {} (str (count subs) " " status)])])]))]]))
+                     (for [[status subs] (sort (seq submissions-count))]
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/tab.cljs]: Does this `seq` do anything?  Looks like it should already be a vector.  And I suppose you're sorting just for consistency's sake?

**Comment by MattPutnam** [comment]
:+1:

**Comment by davidangb** [comment]
updated based on your comments. I thought I needed `reduce + 0` instead of `apply +` to get the default of 0 for an empty collection ... but I guess not!

yes, I'm using sort just for consistency, but I did remove the extra seq.

**Comment by MattPutnam** [comment]
:+1: Good to go

**Pull request merged**